source/workers/LocalWorker.cpp

```diff
@@ -3859,6 +3932,48 @@ void LocalWorker::s3ModeIterateCustomObjects()
 #endif // S3_SUPPORT
 }
 
+/**
+ * Throw an informative WorkerException if the outcome has an error
+ *
+ * @throw WorkerException on error.
+ */
+template <typename OutcomeType>
+inline void throwOnError(const OutcomeType& outcome, const std::string& operation,
+                         const std::string& bucketName)
+{
+    if (!outcome.IsSuccess())
+    {
+        const auto& err = outcome.GetError();
+
+        std::stringstream errStr;
+        errStr << "Unable to " << operation << ": " << err.GetMessage() << std::endl
+               << "Bucket: " << bucketName << std::endl;
+
+        throw WorkerException(errStr.str());
+    }
+}
+
+/**
+ * Throw an informative WorkerException if the outcome has an error
+ *
+ * @throw WorkerException on error.
+ */
+template <typename OutcomeType>
+inline void throwOnError(const OutcomeType& outcome, const std::string& operation,
```

Comment:
nice idea with the throwOnError()!
it's just missing a few extra details like http response code at the moment. here's an example from another funtion:
```
		throw WorkerException(std::string("Getting bucket ACL failed. ") +
			"Endpoint: " + s3EndpointStr + "; "
			"Bucket: " + bucketName + "; "
			"Exception: " + s3Error.GetExceptionName() + "; " +
			"Message: " + s3Error.GetMessage() + "; " +
			"HTTP Error Code: " + std::to_string( (int)s3Error.GetResponseCode() ) );
```